### PR TITLE
Convert users of old failure-domain labels to new

### PR DESCRIPTION
**Pull request by thockin**
This PR holds changes that are pretty obviously correct or have been verified by inspection.  Broken down by commits of varying levels of confidence (for easy review).  Given that we are going to leave the label around for a while, this is less aggressive about purging uses of the old than previous drafts.

/kind cleanup
/kind deprecation

```release-note
NONE
```

### CODE REVIEW

**Comment by andrewsykim** [comment]
Wondering if we should set the "primary" label key to point to the GA labels in the cloud provider? I think this was initially added so we can backfill labels in scenarios where only one may be set.

https://github.com/kubernetes/kubernetes/blob/819ff9b0870506b889e156adc5e66e314f8d148e/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L57-L72

**Comment by andrewsykim** [comment]
Or if for some reason the labels are out of sync and we need to choose one as the source of truth.

**Comment by thockin** [comment]
I can't get the Azure tests to pass and there no useful error, just timeout.  Pushing without those changes.

**Comment by thockin** [comment]
Passes tests without Azure - let's merge this and then figure the rest?

**Comment by aramase** [comment]
> Passes tests without Azure - let's merge this and then figure the rest?

There is a PR open to fix the azure tests - https://github.com/kubernetes/test-infra/pull/19890.

**Comment by fedebongio** [comment]
/cc @liggitt 
Interested on your thoughts

File: pkg/util/node/node.go [hunk 1]
```diff
@@ -187,13 +187,14 @@ func GetZoneKey(node *v1.Node) string {
 		return ""
 	}
 
-	// TODO: prefer stable labels for zone in v1.18
+	// TODO: "failure-domain.beta..." names are deprecated, but will
+	// stick around a long time due to existing on old extant objects like PVs.
+	// Maybe one day we can stop considering them (see #88493).
```
Comment (liggitt) [inline pkg/util/node/node.go:192]: we can still consider them as a fallback, but the goal was to give priority to the GA labels

Comment (thockin) [reply pkg/util/node/node.go:192]: I don't know if that's a good idea - if they are the same value, it doesn't matter.  If they are different values - changing the priority is a breaking change for what gain?  If we are eventually going to have a break, just break it all the way

**Comment by thockin** [comment]
@andrewsykim - looking for an LGTM here, when you get a sec.  This is mostly mechanical now.

**Review by andrewsykim — COMMENTED** [review]
/lgtm

**Pull request merged**